xaitk_saliency/impls/perturb_image/random_grid.py

```diff
@@ -0,0 +1,124 @@
+import numpy as np
+from skimage.transform import resize
+from typing import Optional, Dict, Any, Tuple
+
+from xaitk_saliency import PerturbImage
+from smqtk_descriptors.utils import parallel_map
+
+
+class RandomGrid (PerturbImage):
+    """
+    Generate masks using a random grid of set cell size. If the chosen cell
+    size does not divide an image evenly, then the grid is over-sized and
+    the resulting mask is centered and cropped. Each mask is also shifted
+    randomly by a maximum of half the cell size in both x and y.
+
+    This method is based on RISE (http://bmvc2018.org/contents/papers/1064.pdf)
+    but aims to address the changing cell size, given images of different
+    sizes, aspect of that implementation. This method keeps cell size constant
+    and instead adjusts the overall grid size for different sized images.
+
+    :param n: Number of masks to generate.
+    :param s: Dimensions of the grid cells in pixels. E.g. (3, 4) would use
+        a grid of 3x4 pixel cells.
+    :param p: Probability of a grid cell being set to 1 (not occluded).
```

Comment:
Quick question- should we rename this parameter `p1` (here and below) to be consistent with RISE? I had some confusion whether the probability `p` corresponded to setting a grid to 0 or 1 (in some earlier discussions with Elim), and adding the 1 may help resolve some of this ambiguity.

Reply:
I think that makes sense. I'll update and autosquash.